app.js

```diff
@@ -284,21 +284,42 @@ window.onload = () => {
 };
 
 /* Validate form */
-function onSubmit(event) {
-  const validate = document.getElementById('email');
-  const errorMessage = document.getElementById('error');
-  const email = validate.value;
 
+// Form Validation Section
+
+const form = document.getElementById('form');
+const formEmail = document.getElementById('email');
+const errorMessage = document.getElementById('error');
+const email = formEmail.value;
+form.addEventListener('submit', (e) => {
   if (email.toLowerCase() !== email) {
-    event.preventDefault();
-    validate.classList.add('invalid');
+    e.preventDefault();
+    formEmail.classList.add('invalid');
     errorMessage.classList.add('error');
     errorMessage.innerText = 'Email should be lowercase!';
   } else {
-    validate.classList.remove('invalid');
+    formEmail.classList.remove('invalid');
     errorMessage.classList.remove('error');
+    form.submit();
   }
-}
+});
+// Preserve Storage Section
+const formName = document.getElementById('name');
+const formComment = document.getElementById('textarea');
+
+form.addEventListener('submit', () => {
+  const Data = {
+    Name: formName.value,
+    Email: formEmail.value,
+    Text: formComment.value,
+  };
+  localStorage.setItem('data', JSON.stringify(Data));
+});
 
-const form = document.getElementById('form');
-form.addEventListener('submit', onSubmit);
\ No newline at end of file
+const AutoFiller = localStorage.getItem('data');
+if (AutoFiller) {
+  const DataStored = JSON.parse(localStorage.getItem('data'));
```

Comment:
- [ ] Filled data in the form input are not reflecting in the local storage. Whiles a user is entering a data (_onChange event_), those data need to reflect in the local storage on the fly, as mentioned in the previous review. As you can see from the screenshot below, the entered data in the form are not being persisted to the local storage. Kindly ensure to improve upon the same.

![image](https://user-images.githubusercontent.com/12745474/202845473-7c18ab06-6912-4772-926b-1977510ca824.png)